### PR TITLE
Add local release notes formatting script

### PR DESCRIPTION
**Pull request by BenWilson2**
Adds a script for local generation of release notes formatting

### CODE REVIEW

**Comment by BenWilson2** [comment]
Used the script to generate the links for the 2.14.2 release

File: dev/local-format.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const fs = require('fs');
+const path = require('path');
+
+const { generateLinks } = require('../.github/workflows/format-release-note');
+
+const getMarkdownFiles = (dir) => {
+  let results = [];
+  const list = fs.readdirSync(dir);
```
Comment (daniellok-db) [inline dev/local-format.js:8]: we can provide the `{ withFileTypes: true }` option to return the files as [fs.Dirent](https://nodejs.org/api/fs.html#class-fsdirent) objects, which have the `.isDirectory()` method on them. that way we don't need to do an extra `statSync()` call

```suggestion
  const list = fs.readdirSync(dir, { withFileTypes: true });
```

File: dev/local-format.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  list.forEach(file => {
+    file = path.resolve(dir, file);
+    const stat = fs.statSync(file);
+    if (stat && stat.isDirectory()) {
+      results = results.concat(getMarkdownFiles(file));
+    } else {
+      if (file.endsWith("release.md") && !isPreRelease(file)) {
```
Comment (daniellok-db) [inline dev/local-format.js:15]: after changing to `{ withFileTypes: true }`, you can get the filename with `file.name`

```suggestion
      if (file.name.endsWith("release.md") && !isPreRelease(file.name)) {
```

File: dev/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## Local generation of release notes formatting
+
+If you would like to generate the required links for all release notes in the `website/releases/` directory, you can utilize the 
+dev script located at `dev/local-format.js` as follows:
+
+From the root of the repository, simply run:
+
+```shell
+node dev/local-format.js
```
Comment (daniellok-db) [inline dev/README.md:11]: we could also consider adding this script to `package.json` (https://github.com/mlflow/mlflow-website/blob/main/website/package.json#L5-L18), then people can run it from the `website` directory with `yarn local-format`. the entry would look something like:

```json
"scripts": {
  ...
  "local-format": "node ../dev/local-format.js",
  ...
},
```

**Review by daniellok-db — APPROVED** [review]
Overall LGTM, I think we can consider putting this script in `website/scripts` instead since it looks like that's where the local scripts should be stored. A benefit is that we can use typescript in the script and run it with `ts-node`.

Actually, come to think of it, I should have structured the original workflow like this PR, and set it as a yarn script that the [CI job](https://github.com/mlflow/mlflow-website/blob/main/.github/workflows/ci.yml) can run, rather than have it be a separate job. That would be better, but I think it's out of the scope of this PR, unless you'd like to update the workflow haha.

**Comment by daniellok-db** [comment]
Oh yes and @harupy fixed the issue that was causing the format job to fail on GH actions (the mlflow-automation token expired), so hopefully the CI job works now as well!

File: dev/README.md (hunk 3 above)
Comment (BenWilson2) [reply dev/README.md]: This is a fantastic idea. Migrated and made into a yarn script! (updated dev guide with the script target as well)

File: dev/local-format.js (hunk 1 above)
Comment (BenWilson2) [reply dev/local-format.js]: TIL :D Great tip! Thanks :)

**Pull request merged**